internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/GitDownloader.java

```diff
@@ -0,0 +1,147 @@
+            JSONObject currentReleaseInfo = resultJsonArray.getJSONObject(i);
+            String releaseDate = currentReleaseInfo.get("published_at").toString().substring(0, 10);
+            String version = currentReleaseInfo.getString("tag_name");
+            Long dateDiff;
+            try {
+                dateDiff = getDateDiffFromLastWeeklyRelease(releaseDate);
+            } catch (ParseException e) {
+                throw new DownloaderException("Error occurred while parsing release date.", e);
+            }
+            if (dateDiff <= 8) {
+                File tarDir = null;
+                JSONArray assetsArray = (JSONArray) currentReleaseInfo.get(JSScannerConstants.ASSETS);
+                String name;
+                for (int j = 0, assetSize = assetsArray.length(); j < assetSize; j++) {
+                    JSONObject assetInArray = assetsArray.getJSONObject(j);
+                    //perform exact matching
+                    name = assetInArray.get(JSScannerConstants.NAME).toString();
+                    if ((name.contains(JSScannerConstants.AM) || name.contains(JSScannerConstants.APIM) ||
+                            name.contains(JSScannerConstants.INTEGRATION) ||
+                            name.contains(JSScannerConstants.STREAMPROCESSOR) ||
+                            name.contains(JSScannerConstants.IDENTITYSERVER)) &&
+                            name.endsWith(JSScannerConstants.ZIP_PREFIX)) {
+                        //If current product asset is weekly release the product should be saved in weekly release
+                        // folder of particular product.
+                        //If current product asset is GA release the product should be saved in GA release
+                        //folder of particular product.
+                        if (isWeeklyRelease(name)) {
+                            tarDir = new File(path + File.separator + JSScannerConstants.WEEKLY_RELEASE);
+                        } else if (isGARelease(name)) {
+                            tarDir = new File(path + File.separator + JSScannerConstants.GA_RELEASE);
+                        }
+                        if (tarDir != null) {
+                            try {
+                                createDirectory(tarDir);
+                            } catch (IOException e) {
+                                throw new DownloaderException("Error occurred while " +
+                                        "creating directory for." + name, e);
+                            }
+                        }
+                        log.info(name + " is published on " + releaseDate);
+                        //get download endpoint url
+                        downloadURL = (String) assetInArray.get(JSScannerConstants.DOWNLOAD_URL);
+                        if (downloadURL != null && path != null) {
+                            // the path of downloaded product pack
+                            assert tarDir != null;
```

Comment:
As mentioned in previous comments, don't use asserts. Fix this in all the locations